hello_world_on_chain/test/user.ts

```diff
@@ -1,79 +1,93 @@
     try {
-        const dir = path.resolve( __dirname, jsonPath);
-        const file = fs.readFileSync(dir);
+        const dir: string = path.resolve( __dirname, jsonPath);
+        const file: string = fs.readFileSync(dir, "utf8");
 
-        const data = JSON.parse(file);
+        const data: any = JSON.parse(file);
 
-        const k = data.enclave_attested_application_public_key.claims.public_key.data
-        const pubKeyBytes = ethers.decodeBase64(k)
-        const publicKeyHex = Buffer.from(pubKeyBytes).toString('hex');
+        const k: any =
+            data.enclave_attested_application_public_key.claims.public_key.data
+        const pubKeyBytes: Uint8Array = ethers.decodeBase64(k)
+        const publicKeyHex: string = Buffer.from(pubKeyBytes).toString('hex');
 
-        const taBytes = ethers.decodeBase64(data.transitive_attested_function_call.transitive_attestation)
-        const ta = Buffer.from(taBytes).toString('utf-8');
+        const taBytes:Uint8Array = ethers.decodeBase64(
+            data.transitive_attested_function_call.transitive_attestation
+        )
+        const ta: string = Buffer.from(taBytes).toString('utf-8');
 
-        return {
-            publicKey: `0x${publicKeyHex}`,
-            transitiveAttestation: ta
-        };
+        return { publicKey: `0x${publicKeyHex}`, transitiveAttestation: ta};
     } catch (e) {
-        console.log(`e`, e)
+        throw new Error(`Error loading EVM link data: ` + e);
     }
 }
 
-const loadUserDeployedAddress = () => {
+const loadUserDeployedAddress: () => string  = () : string  => {
     try {
-        const dir = path.resolve(
+        const dir: string = path.resolve(
             __dirname,
             "../deployments/user_deployed_address"
         )
-        const file = fs.readFileSync(dir, "utf8")
+        const file: string = fs.readFileSync(dir, "utf8")
 
         return file.toString()
     } catch (e) {
-        console.log(`e`, e)
+        throw new Error(`loading user deployed address: ` + e);
     }
 }
 
-const loadUserContractABI = () => {
+const loadUserContractABI: () => any = () : any =>  {
     try {
-        const dir = path.resolve(
+        const dir: string = path.resolve(
             __dirname,
             "../artifacts/contracts/User.sol/User.json"
         )
-        const file = fs.readFileSync(dir, "utf8")
-        const json = JSON.parse(file)
+        const file: string = fs.readFileSync(dir, "utf8")
+        const json: any = JSON.parse(file)
         return json.abi
     } catch (e) {
-        console.log(`e`, e)
+        throw new Error(`loading user contract ABI: ` + e);
     }
 }
 
-describe("Local Test", function () {
-    async function deployUser() {
-        const contract = await hre.ethers.deployContract("User");
+interface UserContract extends ethers.Contract {
+    // @ts-ignore
+    processTAHelloWorld(publicKey: any, ta: any): Promise<ethers.ContractTransactionResponse>;
```

Comment:
This is the only Goland error I've been unable to address. I've tried defining methods in a number of different ways but it does not seem to like any of them. So, I've added an "ignore" directive, but if you want I can spend more time trying to fix it.

Reply:
Specifically, it gives the following error:
```
TS2411: Property processTAHelloWorld of type
(publicKey: any, ta: any) => Promise<ContractTransactionResponse>
is not assignable to string index type BaseContractMethod<any[], any, any
```